### PR TITLE
Am/chore/0.6.1 backports

### PR DESCRIPTION
**Pull request by IceTDrinker**
CPU backports

### CODE REVIEW

File: .github/workflows/m1_tests.yml [hunk 1]
```diff
@@ -86,6 +86,13 @@ jobs:
         run: |
           make test_boolean
 
+      # Because we do "illegal" things with the build system which Cargo does not seem to like much
```
Comment (mayeul-zama) [inline .github/workflows/m1_tests.yml:89]: What is it?
Maybe give more info here

File: tfhe/c_api_tests/test_boolean_keygen.c [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#include "tfhe.h"
+#include <tfhe.h>
```
Comment (mayeul-zama) [inline tfhe/c_api_tests/test_boolean_keygen.c:1]: Why this change?
It seems that non system libraries should use ""
https://stackoverflow.com/questions/3162030/difference-between-angle-bracket-and-double-quotes-while-including-heade

File: .github/workflows/m1_tests.yml (hunk 1 above)
Comment (IceTDrinker) [reply .github/workflows/m1_tests.yml:89]: was merged in main like that, this is just a backport

File: tfhe/c_api_tests/test_boolean_keygen.c (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/c_api_tests/test_boolean_keygen.c:1]: I took the dominant style in the tests, I have a fix to make for GPU anyways it seems so will see to integrate

File: .github/workflows/m1_tests.yml (hunk 1 above)
Comment (IceTDrinker) [reply .github/workflows/m1_tests.yml:89]: I don't have the full explanation except it's using a cached build which is not complete see 2 lines below

**Review by mayeul-zama — APPROVED** [review]
